CI is failing

> I'm actually experiencing errors on a fresh clone:
> 
> ```
> ~/Desktop 🦍
> $ git clone https://github.com/remark-embedder/core.git
> Cloning into 'core'...
> remote: Enumerating objects: 186, done.
> remote: Counting objects: 100% (63/63), done.
> remote: Compressing objects: 100% (48/48), done.
> remote: Total 186 (delta 30), reused 26 (delta 10), pack-reused 123
> Receiving objects: 100% (186/186), 65.03 KiB | 924.00 KiB/s, done.
> Resolving deltas: 100% (92/92), done.
> ~/Desktop 🦍
> $ cd core 
> ~/Desktop/core (main) 🦍
> $ npm run setup
> running command with prefix "setup"
> 
> > @remark-embedder/core@0.0.0-semantically-released setup
> > npm install && npm run validate -s
> 
> npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
> npm WARN deprecated object-keys@0.2.0: Please update to the latest object-keys
> npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
> 
> added 1189 packages, and audited 1190 packages in 20s
> 
> 165 packages are looking for funding
>   run `npm fund` for details
> 
> 12 vulnerabilities (6 moderate, 6 high)
> 
> To address issues that do not require attention, run:
>   npm audit fix
> 
> To address all issues (including breaking changes), run:
>   npm audit fix --force
> 
> Run `npm audit` for details.
> [build] Successfully compiled 1 file with Babel (645ms).
> [build] Generating TypeScript definitions
> [typecheck] src/index.ts(92,26): error TS2339: Property 'value' does not exist on type 'StaticPhrasingContent'.
> [typecheck]   Property 'value' does not exist on type 'Emphasis'.
> [typecheck] src/index.ts(97,19): error TS2339: Property 'value' does not exist on type 'Link'.
> [typecheck] src/index.ts(163,32): error TS2339: Property 'tagName' does not exist on type 'Parent'.
> [typecheck] src/index.ts(164,38): error TS2339: Property 'properties' does not exist on type 'Parent'.
> [typecheck] npm run typecheck --silent exited with code 1
> --> Sending SIGTERM to other processes..
> [test] npm run test --silent -- --coverage exited with code SIGTERM
> --> Sending SIGTERM to other processes..
> [build] npm run build --silent exited with code SIGTERM
> --> Sending SIGTERM to other processes..
> [lint] npm run lint --silent exited with code SIGTERM
> NPM command npm run setup -- failed with code 1
> ```

_Originally posted by @kentcdodds in https://github.com/remark-embedder/core/pull/30#issuecomment-881757529_

--- a/package.json
+++ b/package.json
@@ -39,7 +39,8 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.14.6",
-    "@types/mdast": "^3.0.6",
+    "@types/hast": "^2.3.2",
+    "@types/mdast": "^3.0.7",
     "hast-util-from-parse5": "^6.0.1",
     "parse5": "^6.0.1",
     "unified": "^9.2.2",

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,7 +1,8 @@
+import type {Element, Root} from 'hast'
 import fromParse5 from 'hast-util-from-parse5'
 import parse5 from 'parse5'
 import type {Plugin} from 'unified'
-import type {Link, Parent, Text} from 'mdast'
+import type {Link, Paragraph, Text} from 'mdast'
 import visit from 'unist-util-visit'
 
 type GottenHTML = string | null
@@ -46,8 +47,8 @@ type RemarkEmbedderOptions = {
 
 // results in an AST node of type "root" with a single "children" node of type "element"
 // so we return the first (and only) child "element" node
-const htmlToHast = (string: string): Parent =>
-  (fromParse5(parse5.parseFragment(string)) as Parent).children[0] as Parent
+const htmlToHast = (string: string): Element =>
+  (fromParse5(parse5.parseFragment(string)) as Root).children[0] as Element
 
 const getUrlString = (url: string): string | null => {
   const urlString = url.startsWith('http') ? url : `https://${url}`
@@ -74,9 +75,9 @@ const remarkEmbedder: Plugin<[RemarkEmbedderOptions]> = ({
   )
 
   return async tree => {
-    const nodeAndURL: Array<{parentNode: Parent; url: string}> = []
+    const nodeAndURL: Array<{parentNode: Paragraph; url: string}> = []
 
-    visit(tree, 'paragraph', (paragraphNode: Parent) => {
+    visit(tree, 'paragraph', (paragraphNode: Paragraph) => {
       if (paragraphNode.children.length !== 1) {
         return
       }
@@ -89,14 +90,13 @@ const remarkEmbedder: Plugin<[RemarkEmbedderOptions]> = ({
         node.type === 'link' &&
         !node.title &&
         node.children.length === 1 &&
-        node.children[0].value === node.url
-      if (!isText && !isValidLink) {
+        (node.children[0] as Text).value === node.url
+      if (!(isText || isValidLink)) {
         return
       }
 
-      const {url, value = url} = node as Link
-
-      const urlString = getUrlString(value as string)
+      const value = isText ? (node as Text).value : (node as Link).url
+      const urlString = getUrlString(value)
       if (!urlString) {
         return
       }

